RecoTracker/LST/plugins/alpaka/LSTInputProducer.cc

```diff
@@ -0,0 +1,225 @@
+        beamSpotToken_(consumes(iConfig.getParameter<edm::InputTag>("beamSpot"))),
+        lstPixelSeedsPutToken_(produces()),
+        lstInputPutToken_(produces()) {
+    seedTokens_ =
+        edm::vector_transform(iConfig.getParameter<std::vector<edm::InputTag>>("seedTracks"),
+                              [&](const edm::InputTag& tag) { return consumes<edm::View<reco::Track>>(tag); });
+  }
+
+  void LSTInputProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+    edm::ParameterSetDescription desc;
+
+    desc.add<double>("ptCut", 0.8);
+
+    desc.add<edm::InputTag>("phase2OTRecHits", edm::InputTag("siPhase2RecHits"));
+
+    desc.add<edm::InputTag>("beamSpot", edm::InputTag("offlineBeamSpot"));
+    desc.add<std::vector<edm::InputTag>>("seedTracks",
+                                         std::vector<edm::InputTag>{edm::InputTag("lstInitialStepSeedTracks"),
+                                                                    edm::InputTag("lstHighPtTripletStepSeedTracks")});
+
+    descriptions.addWithDefaultLabel(desc);
+  }
+
+  void LSTInputProducer::produce(edm::StreamID iID, device::Event& iEvent, const device::EventSetup& iSetup) const {
+    // Get the phase2OTRecHits
+    auto const& phase2OTHits = iEvent.get(phase2OTRecHitToken_);
+
+    std::vector<unsigned int> ph2_detId;
+    ph2_detId.reserve(phase2OTHits.dataSize());
+    std::vector<float> ph2_x;
+    ph2_x.reserve(phase2OTHits.dataSize());
+    std::vector<float> ph2_y;
+    ph2_y.reserve(phase2OTHits.dataSize());
+    std::vector<float> ph2_z;
+    ph2_z.reserve(phase2OTHits.dataSize());
+    std::vector<TrackingRecHit const*> ph2_hits;
+    ph2_hits.reserve(phase2OTHits.dataSize());
+
+    for (auto const& it : phase2OTHits) {
+      const DetId hitId = it.detId();
+      for (auto const& hit : it) {
+        ph2_detId.push_back(hitId.rawId());
+        ph2_x.push_back(hit.globalPosition().x());
+        ph2_y.push_back(hit.globalPosition().y());
+        ph2_z.push_back(hit.globalPosition().z());
+        ph2_hits.push_back(&hit);
+      }
+    }
+
+    // Get the pixel seeds
+    auto const& mf = iSetup.getData(mfToken_);
+    auto const& bs = iEvent.get(beamSpotToken_);
+
+    // Vector definitions
+    std::vector<float> see_px;
+    std::vector<float> see_py;
+    std::vector<float> see_pz;
+    std::vector<float> see_dxy;
+    std::vector<float> see_dz;
+    std::vector<float> see_ptErr;
+    std::vector<float> see_etaErr;
+    std::vector<float> see_stateTrajGlbX;
+    std::vector<float> see_stateTrajGlbY;
+    std::vector<float> see_stateTrajGlbZ;
+    std::vector<float> see_stateTrajGlbPx;
+    std::vector<float> see_stateTrajGlbPy;
+    std::vector<float> see_stateTrajGlbPz;
+    std::vector<int> see_q;
+    std::vector<std::vector<int>> see_hitIdx;
+    TrajectorySeedCollection see_seeds;
+
+    for (size_t iColl = 0; iColl < seedTokens_.size(); ++iColl) {
+      // Get seed tokens
+      auto const& seedToken = seedTokens_[iColl];
```

Comment:
Looks like this could be
```suggestion
    for (auto const& seedToken : seedTokens_) {
```